### PR TITLE
Replace ament_copyright with local script

### LINKED ISSUE #132: Modify pre-commit config to not require ROS 2
## Description

So you can easily run linters without the need of being inside the docker container a ROS 2 environment.

## Definition of done

- [x] Replace `ament_clang_format` with [clang format](https://github.com/ssciwr/clang-format-wheel#use-from-pre-commit) official hook.
- [x] Replace `ament_clang_tidy` with [clang tidy](https://github.com/pocc/pre-commit-hooks#example-usage).
- [x] Replace `ament_uncrustify` with `clang-format`, format `beluga_amcl` code as needed.
- [x] Remove `ament_cpplint`.
- [x] Replace `ament_flake8` with [flake8](https://github.com/PyCQA/flake8).
  We will need a copy of https://github.com/ament/ament_lint/blob/rolling/ament_flake8/ament_flake8/configuration/ament_flake8.ini if we want to the ROS 2 style exactly (do we want?).
- [x] Replace `pep257` with `flake8-docstrings` (plugin of above hook, also enforces `pep257`)
- [ ] Replace `ament_copyright` with something like https://github.com/leoll2/copyright_notice_precommit.
- [x] Make `clang-tidy` only warn if not in the container (probably the best thing to check is if it's possible to generate the compile commands database), see https://github.com/ekumenlabs/beluga/pull/135#issuecomment-1470806419.

## Additional considerations

Low priority, but I think it would be nice to not need a ros 2 environment to run the pre-commit hooks.

### PR DESCRIPTION
**Pull request by nahueespinosa**
### Proposed changes

Closes #132.

I couldn't find a pre-commit hook that I liked, so I wrote one... It isn't half bad IMO.

#### Type of change

- [ ] 🐛 Bugfix (change which fixes an issue)
- [x] 🚀 Feature (change which adds functionality)
- [ ] 📚 Documentation (change which fixes or extends documentation)

### Checklist

- [x] Lint and unit tests (if any) pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] All commits have been signed for [DCO](https://developercertificate.org/)

### Additional comments

The options I considered were:
- https://github.com/leoll2/copyright_notice_precommit
- https://github.com/sbrunner/hooks

### CODE REVIEW

File: tools/check_copyright.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+limitations under the License.\
+'''
+
+
+class FileContentChecker:
+    """Helper class to check the contents of a file and apply fixes."""
+
+    # Regex pattern that matches a year (YYYY) and a year range (YYYY-ZZZZ).
+    # It has capturing groups for both years (the second one is optional).
+    _YEAR_PATTERN = r'([0-9]{4})(?:(?:-)([0-9]{4}))?'
```
Comment (ivanpauno) [inline tools/check_copyright.py:57]: why is the `-` optional?, shouldn't it be:

```suggestion
    _YEAR_PATTERN = r'([0-9]{4})(?:-([0-9]{4}))?'
```

I may have got confused ...

File: tools/check_copyright.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        """Verify that the file has a valid copyright notice.
+
+        :param template: Template copyright notice.
+        :return: True if a copyright notice was found. False otherwise.
+        """
+        pattern = (
+            # Comment lines and escape special characters in the template...
+            re.escape(self._comment_lines(template.format(year='YYYY', name='NAME')))
+            # Then insert year and name patterns.
+            .replace('YYYY', self._YEAR_PATTERN).replace('NAME', self._NAME_PATTERN)
```
Comment (ivanpauno) [inline tools/check_copyright.py:100]: why combining format and replace and not directly formatting with the patterns?

**Review by ivanpauno — COMMENTED** [review]
I have a couple questions, lgtm

File: tools/check_copyright.py (hunk 1 above)
Comment (nahueespinosa) [reply tools/check_copyright.py]: Good catch! It isn't optional actually, but it is an unnecessary non-capturing group.

Fixed in https://github.com/Ekumen-OS/beluga/pull/240/commits/2ba7d7af002c3a3c61bcabf9a227cee1d31529ac.

File: tools/check_copyright.py (hunk 2 above)
Comment (ivanpauno) [reply tools/check_copyright.py]: I see, thanks for clarifying. I got lost before in where each function call ended.
https://github.com/Ekumen-OS/beluga/commit/ab72795b019ed973811a2086edf16b8001e956a0 LGTM!

**Review by ivanpauno — APPROVED** [review]
LGTM

**Pull request merged**